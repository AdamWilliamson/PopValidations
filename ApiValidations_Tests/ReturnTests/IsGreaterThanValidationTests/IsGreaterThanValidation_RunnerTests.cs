using ApprovalTests;
using FluentAssertions;
using ApiValidations;
using ApiValidations_Tests.TestHelpers;

namespace ApiValidations_Tests.ReturnTests.IsGreaterThanValidationTests;

public class IsGreaterThanApi
{
    public int Get() { return 0; }
}

public class IsGreaterThan_TestingValidator : ApiValidator<IsGreaterThanApi>
{
    public IsGreaterThan_TestingValidator()
    {
        DescribeFunc(x => x.Get()).Return.IsGreaterThan(long.MaxValue);
    }
}

public class IsGreaterThanValidation_RunnerTests
{
    [Fact]
    public void WhenDescribing_ItReturnsTheValidation()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new IsGreaterThan_TestingValidator());

        // Act
        var description = runner.Describe();

        // Assert
        description.Results.Should().HaveCount(1);
        description.Results.Should().HaveCount(ValidatableHelper.GetValidatableCount<IsGreaterThanApi>(ValidatableType.NoExceptions));
        Approvals.VerifyJson(JsonConverter.ToJson(description));
    }
}
