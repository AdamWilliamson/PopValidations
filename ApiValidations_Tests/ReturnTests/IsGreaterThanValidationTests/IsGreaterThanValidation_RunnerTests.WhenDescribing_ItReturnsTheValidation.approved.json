{
    "Results": [
        {
            "Property": "Get()->Int32:Return(Int32)",
            "Outcomes": [
                {
                    "Validator": "IsGreaterThanValidation",
                    "Message": "Must be greater than '9223372036854775807'.",
                    "Values": [
                        {
                            "Key": "value",
                            "Value": "9223372036854775807"
                        }
                    ]
                }
            ],
            "ValidationGroups": []
        },
        {
            "Property": "GetGreaterThan0()->Int32:Return(Int32)",
            "Outcomes": [
                {
                    "Validator": "IsGreaterThanValidation",
                    "Message": "Must be greater than '0'.",
                    "Values": [
                        {
                            "Key": "value",
                            "Value": "0"
                        }
                    ]
                }
            ],
            "ValidationGroups": []
        }
    ]
}